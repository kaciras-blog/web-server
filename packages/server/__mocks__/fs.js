const os = require("os");
const { fs } = require("memfs");

module.exports = fs;

// 解决临时目录不存在的问题。因为通常都会假定系统的临时目录是存在的，但memfs并不自动创建这个目录，
// 这就会造成文件夹不存在问题。
fs.mkdirSync(os.tmpdir(), { recursive: true });
