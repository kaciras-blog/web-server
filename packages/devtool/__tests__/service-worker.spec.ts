import { readFileSync } from "fs";
import { join } from "path";
import { expect, it } from "vitest";
import { resolveFixture, runVite, useTempDirectory } from "./test-utils";
import SWPlugin from "../lib/plugin/service-worker";

const outDir = useTempDirectory();

it("should emit the service worker chunk", async () => {
	await runVite({
		build: {
			outDir,
			rollupOptions: {
				input: "entry-images.js",
			},
		},
		plugins: [
			SWPlugin({ src: resolveFixture("sw.js") }),
		],
	});

	expect(readFileSync(join(outDir, "sw.js"), "utf8")).toMatchSnapshot();
});
