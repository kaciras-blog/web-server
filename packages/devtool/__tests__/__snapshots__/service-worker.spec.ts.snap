// Vitest Snapshot v1

exports[`should emit the service worker chunk 1`] = `
"const o=[\\"visible-off.svg\\",\\"test.png\\",\\"tantrum.gif\\",\\"entry-images.js\\"];for(const s of o)console.log(s);
"
`;
